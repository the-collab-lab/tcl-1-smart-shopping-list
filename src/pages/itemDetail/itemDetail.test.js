import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import ItemDetail from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <ItemDetail />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot on load', () => {
  const tree = TestRenderer.create(
    <BrowserRouter>
      <ItemDetail />
    </BrowserRouter>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
