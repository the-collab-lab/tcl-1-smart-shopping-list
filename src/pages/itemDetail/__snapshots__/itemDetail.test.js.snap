// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot on load 1`] = `
<main
  className="css-vdsylo"
>
  <header
    className="css-527nkd"
  >
    <a
      className="css-gq9b4g"
      href="/"
      onClick={[Function]}
      visualstate="default"
    >
      Go Back
    </a>
    <a
      className="css-gq9b4g"
      href="/"
      onClick={[Function]}
      visualstate="default"
    >
      <h1
        className="css-r73csp"
      >
        iNeedToBuy
      </h1>
    </a>
    <a
      href="http://app.ineedtobuy.xyz/"
      target="_blank"
      visualstate="default"
    >
      See Example
    </a>
  </header>
  <section
    className="css-1xpf03y"
  >
    this is the item detail page, I know you selected a specific item, but we're not there yet. BUT this page **does** have a back button.
  </section>
  <footer
    className="footer css-1i3mvty"
  >
    <a
      className="css-gq9b4g"
      href="/"
      onClick={[Function]}
      visualstate="default"
    >
      My List
    </a>
    <a
      className="css-gq9b4g"
      href="/add-item"
      onClick={[Function]}
      visualstate="default"
    >
      Add Item
    </a>
  </footer>
</main>
`;
